sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/HDFSMetadataLog.scala

```diff
@@ -47,16 +46,16 @@ import org.apache.spark.util.ArrayImplicits._
  * Note: [[HDFSMetadataLog]] doesn't support S3-like file systems as they don't guarantee listing
  * files in a directory always shows the latest files.
  */
-class HDFSMetadataLog[T <: AnyRef : ClassTag](sparkSession: SparkSession, path: String)
-  extends MetadataLog[T] with Logging {
-
-  private implicit val formats = Serialization.formats(NoTypeHints)
+class HDFSMetadataLog[T <: AnyRef: ClassTag](sparkSession: SparkSession, path: String)(
+    private final implicit val manifest: Manifest[T])
+    extends MetadataLog[T]
+    with Logging {
 
-  /** Needed to serialize type T into JSON when using Jackson */
-  private implicit val manifest = Manifest.classType[T](implicitly[ClassTag[T]].runtimeClass)
+  private implicit val formats: Formats = Serialization.formats(NoTypeHints)
 
   // Avoid serializing generic sequences, see SPARK-17372
-  require(implicitly[ClassTag[T]].runtimeClass != classOf[Seq[_]],
+  require(
```

Comment:
Just add a newline? If so, please revert this change

Reply:
reverted